python/activator/middleware_interface.py

```diff
@@ -453,16 +453,21 @@ def _export_skymap_and_templates(self, export, center, detector, wcs, filter):
         # TODO: alternately, we need to extract it from the pipeline? (best?)
         # TODO: alternately, can we just assume that there is exactly
         # one coadd type in the central butler?
-        templates = set(_filter_datasets(
-            self.central_butler, self.butler,
-            "*Coadd",
-            collections=self.template_collection,
-            instrument=self.instrument.getName(),
-            skymap=self.skymap_name,
-            where=template_where,
-            findFirst=True))
-        _log.debug("Found %d new template datasets.", len(templates))
-        export.saveDatasets(templates)
+        try:
+            templates = set(_filter_datasets(
+                self.central_butler, self.butler,
+                "*Coadd",
+                collections=self.template_collection,
+                instrument=self.instrument.getName(),
+                skymap=self.skymap_name,
+                where=template_where,
+                findFirst=True))
+        except _MissingDatasetError as err:
+            _log.warning(err)
+            _log.warning("No templates were found")
```

Comment:
I don't think the double log is needed (and in fact it will probably cause problems with Graphana!). I'd hoped the exception text was already clear enough to identify the problem.

Reply:
I'm also worried about making this `warning`, because right now the pipeline produces *lots* of warnings (e.g., one for every source that's on an edge pixel) and this could get buried. I think `error` might be the lesser evil, even if it does scare anybody running the LATISS pipeline.

Reply:
It's now in `error` level. 

I think this is related to something we may want to think more about and refactor.  In operations, if an incoming image doesn't have templates, we still want the single frame processing done.  So maybe what we need is two sets of pipelines and don't bind them together.